File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        ops_zero_even = 0
        for i in range(0, len(s)):
            if i % 2 == 0:
                if s[i] == '1': ops_zero_even += 1
            else:
                if s[i] == '0': ops_zero_even += 1
                
        ops_zero_odd = 0
        for i in range(0, len(s)):
            if i % 2 == 0:
                if s[i] == '0': ops_zero_odd += 1
            else:
                if s[i] == '1': ops_zero_odd += 1
        
        return min(ops_zero_even, ops_zero_odd)

# string
# time: O(n)
# space: O(1)

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.go
func minOperations(s string) int {
    // odd: 0, even: 1
    opsZeroOdd := 0
    for i := 0; i < len(s); i++ {
        if i % 2 == 0 {
            if s[i] == '1' {
                opsZeroOdd++
            }
        } else {
            if s[i] == '0' {
                opsZeroOdd++
            }
        }
    }

    opsZeroEven := 0
    for i := 0; i < len(s); i++ {
        if i % 2 == 0 {
            if s[i] == '0' {
                opsZeroEven++
            }
        } else {
            if s[i] == '1' {
                opsZeroEven++
            }
        }
    }

    return min(opsZeroOdd, opsZeroEven)
}

// string
// time: O(n)
// space: O(1)